server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -76,10 +79,19 @@ public ConsumerCurator() {
     @Transactional
     @Override
     public Consumer create(Consumer entity) {
+        return super.create(commonCreate(entity), true);
+    }
+
+    @Transactional
+    @Override
+    public Consumer create(Consumer entity, boolean flush) {
+        return super.create(commonCreate(entity), flush);
+    }
+
+    public Consumer commonCreate(Consumer entity) {
```

Comment:
Since the methods in AbstractHibernateCurator are generic, the only reason you'd override them at all in this subclass is to add stuff like commonCreate does. A better way to implement all of this without adding additional overhead would be to remove commonCreate and create, and just add the ensureUUID and validateFacts call to the two-param create method.